File: src/ToeplitzMatrix.java
public class ToeplitzMatrix {
    public boolean isToeplitzMatrix(int[][] matrix) {
        final int rows = matrix.length, columns = matrix[0].length;
        int current;
        for (int column = 0 ; column < columns ; column++) {
            current = matrix[0][column];
            for (int row = 1 ; row < rows && column + row < columns ; row++) {
                if (matrix[row][column + row] != current) return false;
            }
        }
        for (int row = 1 ; row < rows - 1 ; row++) {
            current = matrix[row][0];
            for (int column = 1 ; column < columns && row + column < rows ; column++) {
                if (matrix[row + column][column] != current) return false;
            }
        }
        return true;
    }
}

File: python/toeplitz_matrix.py
from typing import List


class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        columns, rows = len(matrix[0]), len(matrix)
        for column in range(columns - 1):
            current = matrix[0][column]
            for row in range(1, rows):
                if column + row >= columns:
                    break
                if matrix[row][column + row] != current:
                    return False
        for row in range(1, rows - 1):
            current = matrix[row][0]
            for column in range(1, columns):
                if row + column >= rows:
                    break
                if matrix[row + column][column] != current:
                    return False
        return True
